magicgui/_qt/widgets/log_slider.py

```diff
@@ -0,0 +1,32 @@
+"""A Slider with a natural logarithmic scale."""
+import math
+
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import QSlider
+
+
+class QLogSlider(QSlider):
+    """A Slider Widget with a natural logarithmic scale."""
+
+    PRECISION = 1000
+
+    def __init__(self, parent=None):
+        super().__init__(Qt.Horizontal, parent=parent)
+        self.setMinimum(0)
+        self.setMaximum(10)
+
```

Comment:
to make this slightly more flexible, perhaps we should add an attribute `self.base = math.e` ... then convert `math.exp(value)` to `math.pow(self.base, value)` and `math.log(value)` to `math.log(value, base=self.base)`

Reply:
Nice touch, I like this suggestion a lot. Done!